src/ion/liveranges.rs

```diff
@@ -1013,13 +1027,33 @@ impl<'a, F: Function> Env<'a, F> {
                                             operand.pos(),
                                         ),
                                     );
+                                } else if self.func.inst_clobbers(inst).contains(preg) {
```

Comment:
The branches of these two conditionals seem to be the same except for the trace in the previous one (unless I'm missing something). Would it be worth extracting the common part to a utility function to ensure they stay the same?

Reply:
Excellent observation -- thanks for noticing this! I think that they differed more originally but with some simplifications they actually experienced convergent evolution. Merged into one conditional now!